internal/testrunner/runners/pipeline/runner.go

```diff
@@ -158,31 +161,86 @@ func (r *runner) loadTestCaseFile(testCaseFile string) (*testCase, error) {
 	return tc, nil
 }
 
-func (r *runner) verifyResults(testCaseFile string, result *testResult, fieldsValidator *fields.Validator) error {
+func (r *runner) verifyResults(testCaseFile string, config *testConfig, result *testResult, fieldsValidator *fields.Validator) error {
 	testCasePath := filepath.Join(r.options.TestFolder.Path, testCaseFile)
 
+	resultWithoutDynamicFields, err := adjustTestResult(result, config)
+	if err != nil {
+		return errors.Wrap(err, "can't adjust test result")
+	}
+
 	if r.options.GenerateTestResult {
-		err := writeTestResult(testCasePath, result)
+		err := writeTestResult(testCasePath, resultWithoutDynamicFields)
 		if err != nil {
 			return errors.Wrap(err, "writing test result failed")
 		}
 	}
 
-	err := compareResults(testCasePath, result)
+	err = compareResults(testCasePath, resultWithoutDynamicFields)
 	if _, ok := err.(testrunner.ErrTestCaseFailed); ok {
 		return err
 	}
 	if err != nil {
 		return errors.Wrap(err, "comparing test results failed")
 	}
 
+	err = verifyDynamicFields(result, config)
+	if err != nil {
+		return err
+	}
+
 	err = verifyFieldsInTestResult(result, fieldsValidator)
 	if err != nil {
 		return err
 	}
 	return nil
 }
```

Comment:
In this implementation it looks like the written test result will not contain dynamic fields. I think it would be nice if the written test results were exactly the same as we get from the Simulate API, that way the user is not surprised by certain fields are missing from it.

Would it be possible to keep the dynamic fields in the test results, write them as-is? Maybe `compareResults` could exclude dynamic fields in the comparison or something like that?

Reply:
It should be possible, but we need to keep in mind that such fields will change their values every time the results are regenerated.

Reply:
Personally I think that's okay - we see similar behavior with `data.json` files in Beats when they are regenerated and, if anything, I think it will get some assurance that the tests did indeed run and refreshed the files with data from the latest run.

Reply:
Fixed